src/components/Button/Button.tsx

```diff
@@ -27,150 +29,97 @@ export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
 }
+type ButtonComponent = RefForwardingComponentWithAsProps<'button', ButtonProps>
 
-interface CustomButtonProps extends React.HTMLAttributes<HTMLOrSVGElement> {}
-
-export interface RefHandle {
-  container: () => HTMLElement | null
-  button: () => HTMLButtonElement | null
-}
-
-const Button = forwardRef<RefHandle, ButtonProps>(
+const Button: ButtonComponent = forwardRef(
   (
     {
       block,
       className,
       children,
       danger,
       disabled = false,
-      onClick,
       icon,
       iconRight,
       loading = false,
       loadingCentered = false,
       shadow = true,
       size = 'tiny',
-      style,
       type = 'primary',
       htmlType,
       ariaSelected,
       ariaControls,
-      tabIndex,
-      role,
       as,
+      containerRef,
       ...props
-    }: ButtonProps,
+    },
     ref
   ) => {
-    // button ref
-    const containerRef = useRef<HTMLElement>(null)
-    const buttonRef = useRef<HTMLButtonElement>(null)
-
-    useImperativeHandle(ref, () => ({
-      container: () => {
-        return containerRef.current
-      },
-      button: () => {
-        return buttonRef.current
-      },
-    }))
-
     // styles
     const showIcon = loading || icon
 
-    let classes = [ButtonStyles['sbui-btn']]
-    let containerClasses = [ButtonStyles['sbui-btn-container']]
-
-    classes.push(ButtonStyles[`sbui-btn-${type}`])
-
-    if (block) {
-      containerClasses.push(ButtonStyles['sbui-btn--w-full'])
-      classes.push(ButtonStyles['sbui-btn--w-full'])
-    }
-
-    if (danger) {
-      classes.push(ButtonStyles['sbui-btn--danger'])
-    }
-
-    if (shadow && type !== 'link' && type !== 'text') {
-      classes.push(ButtonStyles['sbui-btn-container--shadow'])
-    }
-
-    if (size) {
-      classes.push(ButtonStyles[`sbui-btn--${size}`])
-    }
+    const classes = c8s(
+      ButtonStyles['sbui-btn'],
+      ButtonStyles[`sbui-btn-${type}`],
+      block && ButtonStyles['sbui-btn--w-full'],
+      danger && ButtonStyles['sbui-btn--danger'],
+      shadow &&
+        type !== 'link' &&
+        type !== 'text' &&
+        ButtonStyles['sbui-btn-container--shadow'],
+      size && ButtonStyles[`sbui-btn--${size}`],
+      className,
+      loading && loadingCentered && ButtonStyles[`sbui-btn--text-fade-out`]
+    )
 
-    if (className) {
-      classes.push(className)
-    }
+    const containerClasses = c8s(
+      ButtonStyles['sbui-btn-container'],
+      block && ButtonStyles['sbui-btn--w-full']
+    )
 
-    const iconLoaderClasses = [ButtonStyles['sbui-btn--anim--spin']]
+    const iconLoaderClasses = c8s(
+      ButtonStyles['sbui-btn--anim--spin'],
+      loadingCentered && ButtonStyles[`sbui-btn-loader--center`]
+    )
 
-    if (loadingCentered) {
-      iconLoaderClasses.push(ButtonStyles[`sbui-btn-loader--center`])
-    }
-    if (loading && loadingCentered) {
-      classes.push(ButtonStyles[`sbui-btn--text-fade-out`])
-    }
+    const _icon = icon ? (
```

Comment:
here you could simply use `&&` and if icon is falsy, it will just be null/undefined